doc/user-manual.md

```diff
@@ -0,0 +1,738 @@
+- [Historical pains & motivations](#historical-pains--motivations)
+  - [1. Turning an api call results to csv or a map was not easy](#1-turning-an-api-call-results-to-csv-or-a-map-was-not-easy)
+  - [2. Coding a dhis2 app is sometimes too much work for just a one shot specific task](#2-coding-a-dhis2-app-is-sometimes-too-much-work-for-just-a-one-shot-specific-task)
+  - [3. Combining multiple api calls is hard for exemple in postman.](#3-combining-multiple-api-calls-is-hard-for-exemple-in-postman)
+  - [4. Data can come in/out in various format](#4-data-can-come-inout-in-various-format)
+  - [5. Sometimes we need JS or GIS super power](#5-sometimes-we-need-js-or-gis-super-power)
+  - [6. End user autonomy to re-run the recipe at will](#6-end-user-autonomy-to-re-run-the-recipe-at-will)
+  - [7. Standard recipes reusable accross dhis2](#7-standard-recipes-reusable-accross-dhis2)
+  - [8. Recipes can accept parameters](#8-recipes-can-accept-parameters)
+  - [9. Recipe can return more than a table](#9-recipe-can-return-more-than-a-table)
+  - [10. Accessing multiple dhis2 instances](#10-accessing-multiple-dhis2-instances)
+- [Getting started](#getting-started)
+  - [Install the app](#install-the-app)
+  - [Standard recipes](#standard-recipes)
+    - [Users - Super user, inactive user, never logged in audit](#users---super-user-inactive-user-never-logged-in-audit)
+    - [Users : Create users based on a csv](#users--create-users-based-on-a-csv)
+    - [Coordinates](#coordinates)
+    - [XLSForm - Generate a basic xlsform for a program](#xlsform---generate-a-basic-xlsform-for-a-program)
+    - [XLSForm - generate a xlsform from a DataSet](#xlsform---generate-a-xlsform-from-a-dataset)
+    - [XLSForm - Diff two xlsform](#xlsform---diff-two-xlsform)
+    - [Dataviz - update custom attributes of program indicator](#dataviz---update-custom-attributes-of-program-indicator)
+    - [Your recipe ?](#your-recipe-)
+  - [Specific recipes](#specific-recipes)
+- [Coding tutorial](#coding-tutorial)
+  - [Turn an api call in to a csv](#turn-an-api-call-in-to-a-csv)
+    - [v0.0 get all orgunits](#v00-get-all-orgunits)
+    - [v0.1 get all orgunits for a certain level as parameter](#v01-get-all-orgunits-for-a-certain-level-as-parameter)
+  - [Generate a csv to create users based on the level 3](#generate-a-csv-to-create-users-based-on-the-level-3)
+  - [Synchronous api calls in a loop](#synchronous-api-calls-in-a-loop)
+  - [Generate a json with download prompt](#generate-a-json-with-download-prompt)
+  - [Dhis2 periods](#dhis2-periods)
+  - [Let's update things](#lets-update-things)
+    - [First a readonly version](#first-a-readonly-version)
+    - [Then introduce a "dryRun" mode](#then-introduce-a-dryrun-mode)
+    - [Then test on the first record](#then-test-on-the-first-record)
+    - [Then run on all users](#then-run-on-all-users)
+    - [If the recipe is here to stay](#if-the-recipe-is-here-to-stay)
+  - [We want more](#we-want-more)
+
+
+# Disclaimer
```

Comment:
Maybe add a first "What is TaskR" with like "TaskR is a way to create and run scripts called "recipes" that can interacts with the DHIS2 api and generate table, maps or various type of output.